cpp/src/arrow/python/arrow_to_python.cc

```diff
@@ -195,6 +195,7 @@ Status ReadSerializedPythonSequence(std::shared_ptr<io::RandomAccessFile> src,
   RETURN_NOT_OK(ipc::RecordBatchStreamReader::Open(src, &reader));
   RETURN_NOT_OK(reader->ReadNextRecordBatch(batch_out));
   RETURN_NOT_OK(src->Tell(&offset));
+  offset += 4; // TODO(pcm): Why is this neccessary?
```

Comment:
@wesm Could you have a quick look and see why this seems to be neccessary? Am I not closing one of the Readers/Writers correctly or is there some EOS marker written that is not read?